src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```
@@ -0,0 +1,52 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+
+    @Override
+    public List<FruitTransaction> convertStringsToTransactionsList(List<String> lines) {
+        if (lines == null) {
+            throw new NullPointerException("Input parameter cannot be null");
+        }
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        for (int i = 1; i < lines.size(); i++) {
```

You skip the first line by starting iteration at index 1, but you do not validate that the header equals the expected CSV header `type,fruit,quantity`. The task explicitly requires validating the header ("Skip or throw on invalid header or malformed lines (e.g., wrong number of fields, non-integer quantity, empty fruit name, unknown operation code); must not silently ignore without validation"). Please check the header string and throw an informative RuntimeException if it differs from the expected header. Current iteration start: `for (int i = 1; i < lines.size(); i++)` ; header requirement referenced in task description .